File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
       let min = prices[0];
       let max = min;
       let profit = 0;
       for (let i = 0; i < prices.length; i ++) {
        if (prices[i] != prices.length - 1 && prices[i] <= min) {
            min = max = prices[i]
        } else if (prices[i] > max) {
            max = prices[i];
            profit = Math.max(profit, max - min)
        }
       }

       return profit
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-4.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0 

        for i in range(len(prices) - 1): 
            curD = prices[i] 
            nextD = prices[i + 1] 

            profit += max(0, nextD - curD) 

        return profit